core/templates/components/ck-editor-helpers/ck-editor-4-widgets.initializer.ts

```diff
@@ -151,16 +154,31 @@ export class CkEditorInitializerService {
                   },
                   function(widgetShouldBeRemoved) {
                     if (widgetShouldBeRemoved || that.data.isCopied) {
+                      const highlighted = customizationArgSpecs
+                        .some(function(spec) {
+                          return spec.default_value_obtainable_from_highlight;
+                        });
+
                       that.data.isCopied = false;
                       var newWidgetSelector = (
                         '[data-cke-widget-id="' + that.id + '"]');
-                      if (newWidgetSelector !== null) {
-                        var widgetElement = editor.editable().findOne(
-                          newWidgetSelector);
-                        if (widgetElement) {
-                          widgetElement.remove();
-                          editor.fire('change');
-                        }
+                      if (newWidgetSelector === null) {
+                        return;
+                      }
+                      var widgetElement = editor.editable().findOne(
+                        newWidgetSelector);
+
+                      if (!widgetElement) {
+                        return;
+                      }
+
+                      if ((!highlighted && componentIsNewlyCreated) ||
```

Comment:
If you decide to leave this condition, please explain it in a comment, it's not obvious what happens here.

Reply:
Done